source/common/http/http1/codec_impl.cc

```diff
@@ -366,20 +366,16 @@ void ResponseEncoderImpl::encodeHeaders(const ResponseHeaderMap& headers, bool e
 
 static const char REQUEST_POSTFIX[] = " HTTP/1.1\r\n";
 
-void RequestEncoderImpl::encodeHeaders(const RequestHeaderMap& headers, bool end_stream) {
+Status RequestEncoderImpl::encodeHeaders(const RequestHeaderMap& headers, bool end_stream) {
+  // Required headers must be present. This can only happen by some erroneous processing after the
+  // downstream codecs decode.
+  RETURN_IF_ERROR(HeaderUtility::checkRequiredHeaders(headers));
```

Comment:
If filter manager abort with an error when required headers are missing, this call then should always be successful, right?

Reply:
Yes, but after I fixed with just per-iteration checks in the FM, this was brought up as a concern because it requires the caller to perform the check and might not be robust to other extension points. 

So I changed to Status, but also kept the FM per-iteration check since I think it's necessary. (Filter A removes :path, Filter B assumes :path is not nullptr)

Reply:
On the other hand, we can take a posture where filters should not assume required headers are present and always defensively use safe accessors like `getPathValue`. That would leave us with just the status check in encoding.

Reply:
Yeah, I think I was of the mind we were just avoiding the crash on exit codec (which is why I suggested reverting the above) and filters should use getPathValue.
I'm good either way though given it's a constant time check hence marking my own version of this question resolved :-)